From 10a3dc5586f73d3c973908fbba9ddb146b69edb1 Mon Sep 17 00:00:00 2001
From: Cy Schubert <cy@FreeBSD.org>
Date: Thu, 23 Feb 2023 09:45:57 -0800
Subject: [PATCH 20/41] Fix build on FreeBSD.

Now that FreeBSD has imported LLVM15, new errors are flagged.

Ideally the code that implicily converts integers should be fixed but
a first step is to use a compiler option that ignores the errors like
prior versions of LLVM and GCC do.

Additionaly, as in Linux, read_common_sizet() is also undefined in
FreeBSD.
---
 CMakeLists.txt        | 2 +-
 src/common/memusage.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6a9d6828a..1b47e857c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -516,7 +516,7 @@ if (MSVC)
 endif()
 
 if (${CMAKE_C_COMPILER_ID} STREQUAL "GNU" OR ${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pipe -W -Wall -Wnonnull -Wshadow -Wformat -Wundef -Wno-unused-parameter -Wmissing-prototypes -Wno-unknown-pragmas -D_GNU_SOURCE -std=c99")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pipe -W -Wall -Wnonnull -Wshadow -Wformat -Wundef -Wno-unused-parameter -Wmissing-prototypes -Wno-unknown-pragmas -Wno-int-conversion -D_GNU_SOURCE -std=c99")
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
 	add_link_options(-lkvm -lm -lprocstat)
diff --git a/src/common/memusage.c b/src/common/memusage.c
index 01221f888..fbf89340b 100644
--- a/src/common/memusage.c
+++ b/src/common/memusage.c
@@ -56,7 +56,7 @@
 #include "memusage.h"
 #include "bfind.h"
 
-#if defined(OS_LINUX)
+#if defined(OS_LINUX) || defined(__FreeBSD__)
 static int read_common_sizet(void *szp, char *strval)
 {
 	char *end;
-- 
2.34.1

